### PR TITLE
[16.0][IMP] base_edifact: add support for create segment and interchange based on pydifact library

### PR DESCRIPTION
**Pull request by thienvh332**

### CODE REVIEW

File: base_edifact/models/edifact.py [hunk 1]
```diff
@@ -242,3 +242,47 @@ def map2odoo_description(self, seg):
             description = seg[2][3]
             return description
         return None
+
+    def _replace_empty(self, value):
```
Comment (simahawk) [inline base_edifact/models/edifact.py]:
```suggestion
    def _safe_segment_element(self, value):
```

File: base_edifact/models/edifact.py [hunk 2]
```diff
@@ -242,3 +242,47 @@ def map2odoo_description(self, seg):
...
+        for element in elements:
+            if isinstance(element, list):
+                cleaned_elements.append(
+                    [self._replace_empty(value) for value in element]
+                )
+            else:
+                cleaned_elements.append(self._replace_empty(element))
+        return Segment(*cleaned_elements)
+
+    def create_interchange(self, sender, recipient, control_ref, syntax_identifier):
```
Comment (simahawk) [inline base_edifact/models/edifact.py:263]: The docstring should state what the method does

https://peps.python.org/pep-0257/#multi-line-docstrings

**Review by simahawk — APPROVED** [review]
LG, pre-approving

File: base_edifact/models/edifact.py (hunk 1 above)
Comment (thienvh332) [reply base_edifact/models/edifact.py]: Thank you @simahawk for your review. I updated it.

File: base_edifact/models/edifact.py (hunk 2 above)
Comment (thienvh332) [reply base_edifact/models/edifact.py:263]: Thank you @simahawk for your review. I updated it.

**Review by rmorant — APPROVED** [review]

**Review by simahawk — APPROVED** [review]
LG

**Pull request merged**